app/Filament/Resources/ServerResource/Pages/EditServer.php

```diff
@@ -659,7 +664,37 @@ public function form(Form $form): Form
                                     ->deletable(false)
                                     ->addable(false)
                                     ->columnSpan(4),
-                            ])->columns(4),
+                                Forms\Components\Actions::make([
+                                    Action::make('createDatabase')
+                                        ->disabled(fn (Server $server) => DatabaseHost::query()->count() < 1 || $server->databases()->count() >= $server->database_limit)
+                                        ->label(fn (Server $server) => $server->databases()->count() >= $server->database_limit ? 'Database Limit Reached' : (DatabaseHost::query()->count() < 1 ? 'No Database Hosts' : 'Create Database'))
+                                        ->color(fn (Server $server) => $server->databases()->count() >= $server->database_limit || DatabaseHost::query()->count() < 1 ? 'danger' : 'primary')
+                                        ->modalSubmitActionLabel('Create')
+                                        ->action(function (array $data, DatabaseManagementService $service, Server $server) {
+                                            if (empty($data['database'])) {
+                                                $data['database'] = str_random(12);
```

Comment:
Use the `generateUniqueDatabaseName` method from the service the generate the name.